commit 4d60ce8b6000774d6b4a98ee7bffc9388021eb6d
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:22:03 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-ui_info-in-info.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index d0c0315ed..e6c7a9f00 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -403,6 +403,13 @@ proc ui_message {priority prefix args} {
        }
     } 
 
+    # adopt msg priority if the current phase equals the current priority
+    # this can only happen for ui_info during `port info` (currently).
+    if {[info exists current_phase]
+        && ${current_phase} eq ${priority}} {
+        set priority "msg"
+    }
+
     foreach chan $channels($priority) {
         if {[lindex $args 0] eq "-nonewline"} {
             puts -nonewline $chan $prefix[lindex $args 1]
diff --git a/src/port/port.tcl b/src/port/port.tcl
index 9fcd144ea..5b1ba2e13 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -1709,6 +1709,7 @@ proc action_info { action portlist opts } {
         return 1
     }
 
+    set_phase "info"
     set separator ""
     global global_variations global_options
     set gvariations [dict create {*}[array get global_variations]]
